develop: random failure when starting the registry

##### **Overview of the issue**

Only in the current **develop** branch, since this PR https://github.com/jhipster/jhipster-registry/pull/180 by @dajay, when starting the app, it can fail randomly.

Here the stacktrace:

```
2017-10-07 00:59:06.306  WARN 26669 --- [           main] c.n.c.sources.URLConfigurationSource     : No URLs will be polled as dynamic configuration sources.
2017-10-07 00:59:06.319  INFO 26669 --- [           main] com.netflix.discovery.DiscoveryClient    : Shutting down DiscoveryClient ...
2017-10-07 00:59:06.319  INFO 26669 --- [           main] com.netflix.discovery.DiscoveryClient    : Completed shut down of DiscoveryClient
2017-10-07 00:59:06.352 ERROR 26669 --- [           main] o.s.boot.SpringApplication               : Application startup failed

org.springframework.context.ApplicationContextException: Failed to start bean 'documentationPluginsBootstrapper'; nested exception is java.lang.IllegalArgumentException: Multiple entries with same key: springfox.documentation.spring.web.plugins.PathAndParametersEquivalence@438aaa9f.wrap(springfox.documentation.spring.web.WebMvcRequestHandler@3f06ebe0)=[springfox.documentation.spring.web.WebMvcRequestHandler@3f06ebe0] and springfox.documentation.spring.web.plugins.PathAndParametersEquivalence@438aaa9f.wrap(springfox.documentation.spring.web.WebMvcRequestHandler@1752e45c)=[springfox.documentation.spring.web.WebMvcRequestHandler@1752e45c]
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:178)
	at org.springframework.context.support.DefaultLifecycleProcessor.access$200(DefaultLifecycleProcessor.java:50)
	at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:348)
	at org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:151)
	at org.springframework.context.support.DefaultLifecycleProcessor.onRefresh(DefaultLifecycleProcessor.java:114)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:880)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:144)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303)
	at io.github.jhipster.registry.JHipsterRegistryApp.main(JHipsterRegistryApp.java:74)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.WarLauncher.main(WarLauncher.java:59)
Caused by: java.lang.IllegalArgumentException: Multiple entries with same key: springfox.documentation.spring.web.plugins.PathAndParametersEquivalence@438aaa9f.wrap(springfox.documentation.spring.web.WebMvcRequestHandler@3f06ebe0)=[springfox.documentation.spring.web.WebMvcRequestHandler@3f06ebe0] and springfox.documentation.spring.web.plugins.PathAndParametersEquivalence@438aaa9f.wrap(springfox.documentation.spring.web.WebMvcRequestHandler@1752e45c)=[springfox.documentation.spring.web.WebMvcRequestHandler@1752e45c]
	at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:150)
	at com.google.common.collect.RegularImmutableMap.checkNoConflictInBucket(RegularImmutableMap.java:104)
	at com.google.common.collect.RegularImmutableMap.<init>(RegularImmutableMap.java:70)
	at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:254)
	at com.google.common.collect.ImmutableListMultimap.copyOf(ImmutableListMultimap.java:268)
	at com.google.common.collect.ImmutableMultimap.copyOf(ImmutableMultimap.java:298)
	at com.google.common.collect.ImmutableMultimap$Builder.build(ImmutableMultimap.java:272)
	at com.google.common.collect.ImmutableListMultimap$Builder.build(ImmutableListMultimap.java:224)
	at com.google.common.collect.Multimaps.index(Multimaps.java:1511)
	at com.google.common.collect.Multimaps.index(Multimaps.java:1455)
	at springfox.documentation.spring.web.plugins.DefaultRequestHandlerCombiner.combined(DefaultRequestHandlerCombiner.java:59)
	at springfox.documentation.spring.web.plugins.DefaultRequestHandlerCombiner.combine(DefaultRequestHandlerCombiner.java:49)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.defaultContextBuilder(DocumentationPluginsBootstrapper.java:106)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.buildContext(DocumentationPluginsBootstrapper.java:91)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.start(DocumentationPluginsBootstrapper.java:154)
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:175)
	... 20 common frames omitted
```

##### **Reproduce the error**

- checkout the develop branch
- package the application: `./mvnw clean package -Pprod -DskipTests`
- launch the registry: `jar -jar target/*.war`

##### **Related issues**

It seems to be similar to this ticket: https://github.com/springfox/springfox/issues/2039

##### **Suggest a Fix**

This fix https://github.com/jhipster/jhipster-registry/pull/184 doesn't work.

At work, with @PierreBesson, we switch back to Camden.SR7 instead of Dalston, but I don't like this solution.

##### **JHipster Registry Version(s)**

develop branch

--- a/src/main/resources/config/application.yml
+++ b/src/main/resources/config/application.yml
@@ -92,7 +92,7 @@ eureka:
 
 jhipster:
     swagger:
-        default-include-pattern: /(api|config)/.*
+        default-include-pattern: /(api)/.*
         title: JHipster Registry API
         description: JHipster Registry API documentation
         version: ${info.project.version}

